actors/evm/src/interpreter/precompiles/bls_util.rs

```diff
@@ -0,0 +1,421 @@
+use super::PrecompileError;
+use blst::{
+    // Basic types
+    blst_bendian_from_fp,
+    blst_fp,
+    blst_fp_from_bendian,
+    blst_fp2,
+    blst_p1,
+    blst_p1_affine,
+    blst_p1_affine_in_g1,
+    blst_p1_affine_is_inf,
+    blst_p1_affine_on_curve,
+    blst_p1_from_affine,
+    blst_p1_mult,
+    blst_p1_to_affine,
+    blst_p2,
+    blst_p2_affine,
+    blst_p2_affine_in_g2,
+    blst_p2_affine_is_inf,
+    blst_p2_affine_on_curve,
+    blst_p2_from_affine,
+    blst_p2_mult,
+    blst_p2_to_affine,
+    blst_scalar,
+    blst_scalar_from_bendian,
+};
+
+pub const G1_INPUT_LENGTH: usize = 128;
+pub const G1_ADD_INPUT_LENGTH: usize = G1_INPUT_LENGTH * 2;
+pub const G1_OUTPUT_LENGTH: usize = 128;
+pub const PADDING_LENGTH: usize = 16;
+pub const G1_MSM_INPUT_LENGTH: usize = 160;
+pub const SCALAR_LENGTH: usize = 32;
+pub const G2_ADD_INPUT_LENGTH: usize = 512;
+pub const G2_INPUT_ITEM_LENGTH: usize = 256;
+pub const G2_OUTPUT_LENGTH: usize = 256;
+pub const G2_MSM_INPUT_LENGTH: usize = 288;
+pub const PADDED_FP_LENGTH: usize = 64;
+pub const PADDED_FP2_LENGTH: usize = 2 * PADDED_FP_LENGTH;
+pub const PADDED_G1_LENGTH: usize = 2 * PADDED_FP_LENGTH;
+pub const PADDED_G2_LENGTH: usize = 2 * PADDED_FP2_LENGTH;
+pub const PAIRING_INPUT_LENGTH: usize = PADDED_G1_LENGTH + PADDED_G2_LENGTH;
+
+/// FP_LENGTH specifies the number of bytes needed to represent an
+/// Fp element. This is an element in the base field of BLS12-381.
+///
+/// Note: The base field is used to define G1 and G2 elements.
+pub const FP_LENGTH: usize = 48;
+
+/// SCALAR_LENGTH_BITS specifies the number of bits needed to represent an Fr element.
```

Comment:
It's not your fault since it comes from the EIP but it is really confusing that we use both Fr and Fq i.e. [here](https://github.com/filecoin-project/builtin-actors/pull/1669/files#diff-da47f52af3deab18cfd2f0207d5d17fb803367eb6b047d815170c0a740f05da3R105) Please let's be consistent and pick one

Reply:
Ha yeah its really not your fault this notation problem is in revm source: https://github.com/bluealloy/revm/blob/6697f70dd3165e5149a0342b29d6f23f7b082cac/crates/precompile/src/bls12_381_const.rs#L132.  I'm going to recommend just changing this to Fq everywhere since that is what most of the revm code does and is the FIPs notation